src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeadersNonValidated.cs

```diff
@@ -82,10 +82,25 @@ public bool TryGetValues(string headerName, out HeaderStringValues values)
 
         /// <summary>Gets an enumerator that iterates through the <see cref="HttpHeadersNonValidated"/>.</summary>
         /// <returns>An enumerator that iterates through the <see cref="HttpHeadersNonValidated"/>.</returns>
-        public Enumerator GetEnumerator() =>
-            _headers is HttpHeaders headers && headers.GetEntriesArray() is HeaderEntry[] entries ?
-                new Enumerator(entries, headers.Count) :
-                default;
+        public Enumerator GetEnumerator()
+        {
+            if (_headers != null)
+            {
+                if (_headers is HttpContentHeaders contentHeaders)
+                {
+                    _ = contentHeaders.ContentLength;
+                }
```

Comment:
Since you've already created the `TryLoadContentLength` helper, you might as well make it internal and use it here as well.